server/src/main/java/org/opensearch/indices/recovery/RecoverySettings.java

```diff
@@ -177,7 +177,14 @@ public class RecoverySettings {
     );
 
     // choose 512KB-16B to ensure that the resulting byte[] is not a humongous allocation in G1.
-    public static final ByteSizeValue DEFAULT_CHUNK_SIZE = new ByteSizeValue(512 * 1024 - 16, ByteSizeUnit.BYTES);
+    public static final Setting<ByteSizeValue> INDICES_RECOVERY_CHUNK_SIZE_SETTING = Setting.byteSizeSetting(
+        "indices.recovery.chunk_size",
+        new ByteSizeValue(512 * 1024 - 16, ByteSizeUnit.BYTES),
+        new ByteSizeValue(1, ByteSizeUnit.BYTES),
+        new ByteSizeValue(100, ByteSizeUnit.MB),
+        Property.Dynamic,
+        Property.NodeScope
+    );
```

Comment:
@dblock In general, you have to look at the report from the `:server:japicmp` task and see what the incompatibility is. In this case, a public constant has been removed, so the work around is to keep the same public constant visible in the `@PublicApi`-annotated class:

```suggestion
    public static final ByteSizeValue DEFAULT_CHUNK_SIZE = new ByteSizeValue(512 * 1024 - 16, ByteSizeUnit.BYTES);

    public static final Setting<ByteSizeValue> INDICES_RECOVERY_CHUNK_SIZE_SETTING = Setting.byteSizeSetting(
        "indices.recovery.chunk_size",
        DEFAULT_CHUNK_SIZE,
        new ByteSizeValue(1, ByteSizeUnit.BYTES),
        new ByteSizeValue(100, ByteSizeUnit.MB),
        Property.Dynamic,
        Property.NodeScope
    );
```